### ISSUE TITLE
Invalid checksum for markup

### ISSUE THREAD

**Issue by stevenornelas** [codefoundries/isomorphic-material-relay-starter-kit#3]
When I run locally in dev mode I get the following in the browser console:

```
React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server
```

When loading the initial page the screen visibly flashes as it redraws.

**Comment by thelordoftheboards** [comment]
I am able to reproduce the issue. I found the following issue https://github.com/callemall/material-ui/issues/2336 on the Material-UI repository, I will look into implementing some of the ideas there.

**Comment by thelordoftheboards** [comment]
I was able to implement the solution proposed by https://github.com/chadrien and it works well.

**Comment by bfwg** [comment]
Hi, I have tried @thelordoftheboards solution and it is not working, my warning is:
```
Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) <noscript data-reacti
 (server) <div data-reactid=".1
```
Is there a way to solve this?

Thanks

**Comment by thelordoftheboards** [comment]
This has been an on and off problem. After some changes I was able to make it better, but not always.
Since React 15 will bring big changes to what will be in the actual DOM, namely the react id is going away, I figured out I would wait for it to come out, wait for whatever accompanying changes in Material-UI get released, and then look at it again. I will keep this open for now as a placeholder. If you figure it out in the meantime a PR would be greatly appreciated.
